presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSplitManager.java

```diff
@@ -630,10 +643,23 @@ public Map<String, PartitionStatistics> getPartitionStatistics(MetastoreContext
     private static class TestingDirectoryLister
             implements DirectoryLister
     {
+        private final ConnectorSession connectorSession;
+
+        public TestingDirectoryLister(ConnectorSession connectorSession)
+        {
+            this.connectorSession = connectorSession;
+        }
+
         @Override
         public Iterator<HiveFileInfo> list(ExtendedFileSystem fileSystem, Table table, Path path, Optional<Partition> partition, NamenodeStats namenodeStats, HiveDirectoryContext hiveDirectoryContext)
         {
             try {
+                Map<String, String> additionalProperties = hiveDirectoryContext.getAdditionalProperties();
+                assertEquals(additionalProperties.get(PRESTO_QUERY_ID), connectorSession.getQueryId());
+                assertEquals(Optional.ofNullable(additionalProperties.get(PRESTO_QUERY_SOURCE)), connectorSession.getSource());
+                assertEquals(Optional.ofNullable(additionalProperties.get(PRESTO_CLIENT_INFO)), connectorSession.getClientInfo());
+                assertEquals(additionalProperties.get(PRESTO_USER_NAME), connectorSession.getUser());
+                assertEquals(Optional.ofNullable(additionalProperties.get(PRESTO_METASTORE_HEADER)), getMetastoreHeaders(connectorSession));
```

Comment:
Should this go in a different test instead of leveraging this inside the TestingDirectoryLister ?